prow/config/config.go

```diff
@@ -174,6 +180,35 @@ func (c *Config) InRepoConfigEnabled(identifier string) bool {
 	return false
 }
 
+// InRepoConfigAllowsCluster determines if a given cluster may be used for a given repository
+func (c *Config) InRepoConfigAllowsCluster(clusterName, repoIdentifier string) bool {
+	if allowedClusters, exists := c.InRepoConfig.AllowedClusters[repoIdentifier]; exists {
```

Comment:
They are needed for the currently tested behaviour of "do not fall through to the next layer if not allowed", i.E. if there is something like
```
"org": ["bar"]
"org/repo": ["foo"]
```

then `org/repo` is not allowed to use the `bar` cluster. Thats what I first thought of as being intuitive but on second thought I am not sure anymore. WDYT?

Reply:
I would expect unioning to make more sense than overriding for this use case, but I'm not certain. I think either way will be fine, but unioning seems like it would be better for concise config.